[리액트 8기 이재윤] 5주차 과제 1

jaeyooniverse opened the pull request:
commit log 정확히 지켜가며 과제 진행해보겠습니다!

jaeyooniverse committed:
기본 세팅

jaeyooniverse committed:
reducer 테스트

jaeyooniverse committed:
reducer 작성

jaeyooniverse committed:
action 테스트

jaeyooniverse commented:
action 테스트를 작성하고 Green Cycle을 진행하는 중에 분명 맞게 작성한 거 같은데 계속 fail이 나서 (분명 함수로 작성하고 함수로 호출해도 함수가 아니라고 하는 에러) 먼저 제출하신 분들의 코드를 참고하다 redux-mock-store를 이용하면 redux-thunk 테스트를 쉽게 할 수 있다는 정보를 얻어서 redux-mock-store를 설치한 후 action 테스트를 수정했습니다!

jaeyooniverse commented:
복무지에서 맥북으로 작업하다 집에 와서 컴퓨터로 작성하던 중에 git 설정을 잘못해서 집에서 하던 내용이 싹 다 날아가 버렸습니다.. ㅠㅠ 아무튼 reducer, action Red - Green - Refactor까지 다 했음을 알려드립니다... 내일 다시 가서 작업하겠습니다 ㅜㅠ

jaeyooniverse committed:
actions 테스트, 코드 Refactoring

hannut91 commented:
commit log를 제대로 작성하려고 하시니, commit에 대해서도 피드 포워드를 드려야겠네요.

1. 커밋의 본문이 없습니다. 커밋의 첫 줄에는 제목이, 그리고 한 줄 띄고 다음 줄부터 커밋의 본문이 옵니다. 
2. 제목에는 지금은 재윤님이 한 일을 제목에 적고 계십니다. 그 마저도 무슨 일을 했는지 잘 모르겠습니다. 제목에는 커밋의 의도, 이유가 들어가야 합니다. 왜 이 작업을 했는지가 드러나야 합니다. 그리고 내용에는 그 이유를, 의도를 위해서 실제로 한 일을 아주 구체적으로 적어주셔야 합니다.

위의 조언을 보고 다시 한 번 해보세요. 그리고 다시 피드 포워드를 드릴게요

hannut91 commented on src/actions.js:
```diff
@@ -0,0 +1,66 @@
+  };
+}
+
+export function loadRegions() {
+  return async (dispatch) => {
+    const regions = await fetchRegions();
+
+    dispatch(setRegions(regions));
+  };
+}
+
+export const setCategories = (categories) => ({
+  type: 'setCategories',
+  payload: {
+    categories,
+  },
+});
```
function가 arrow function이 같이 사용되고 있네요. 둘 중 하나로 일관성있게 사용하는 것은 어떨까요?

jaeyooniverse replied:
arrow function으로 모두 리팩토링 했습니다!
done: https://github.com/CodeSoom/react-week5-assignment-1/pull/98/commits/987e87b3ee4a81accdaf0bcf0f0d525ebe894a31

hannut91 commented on src/actions.test.js:
```diff
@@ -0,0 +1,55 @@
+
+describe('actions', () => {
+  describe('loadRegions', () => {
+    it('"loadRegions"가 dispatch되면 "setRegions" 액션을 dispatch한다', async () => {
```
`"loadRegions"가 dispatch되면`은 없어도 될 것 같아요

jaeyooniverse replied:
제거했습니다.
done: https://github.com/CodeSoom/react-week5-assignment-1/pull/98/commits/a7e5a62e1b141efe830d6f0ac97c03239cecfee6

hannut91 commented on src/reducer.js:
```diff
@@ -0,0 +1,70 @@
+  };
+};
+
+const actionTypes = {
+  setRegions: (state, action) => setRegions(state, action),
+  setCategories: (state, action) => setCategories(state, action),
+  setRestaurants: (state, action) => setRestaurants(state, action),
+  changeRegionName: (state, action) => changeRegionName(state, action),
+  changeCategoryId: (state, action) => changeCategoryId(state, action),
+};
```
```js

const actionTypes = {
  setRegions,
  setCategories,
  setRestaurants,
  changeRegionName,
  changeCategoryId,
};
```

이렇게 쓰셔도 됩니다

jaeyooniverse replied:
수정했습니다!
done: https://github.com/CodeSoom/react-week5-assignment-1/pull/98/commits/3def91fd2baa2ce8ae4afc692e76b6e0d151cbdd

hannut91 commented on src/reducer.test.js:
```diff
@@ -0,0 +1,101 @@
+} from './actions';
+
+describe('reducer', () => {
+  context('존재하는 액션 타입이면', () => {
+    describe('setRegions', () => {
+      it('sets regions with new regions', () => {
+        const regions = [{ id: 1, name: '수원' }];
+
+        const initialState = {
+          regions: [],
+        };
+
+        const state = reducer(initialState, setRegions(regions));
+
+        expect(state.regions).toHaveLength(1);
+      });
+    });
```
구조가 뭔가 이상한구요. 안에 `describe`는 없어도 될 것 같아요

jaeyooniverse replied:
수정완료
done: https://github.com/CodeSoom/react-week5-assignment-1/pull/98/commits/156e5fd96e816a378b18b8eab9994e9c8db9a594

hannut91 commented on src/reducer.test.js:
```diff
@@ -0,0 +1,101 @@
+} from './actions';
+
+describe('reducer', () => {
+  context('존재하는 액션 타입이면', () => {
```
존재하는 액션과 존재하지 않는 액션을 나누어 주셨어요. 아마도 `reducer`의 동작이 존재하는 타입과 존재하지 않는 타입의 경우 다르게 동작하는 것을 보여주고 싶으셨던 것 같아요.

하지만 존재하는 액션 타입과 존재하지 않는지는 중요한 정보는 아닌 것 같아요. 단지 존재하지 않는 액션 타입이 특별한 예외인거죠. 이 하나의 특별한 예외, 그리고 심지어 많이 쓰이지도 않는 이 예외 때문에 일단 코드의 indent가 늘어났고, 코드를 읽는 사람은 이 것을 계속 머릿속에 담은 채로 코드를 보게 됩니다.

따라서 존재하지 않는 액션의 경우를 그냥 따로 테스트를 명시해주는게 좋을 것 같아요. Context로 바라보는게 아니라요

jaeyooniverse replied:
수정했습니다!
done: https://github.com/CodeSoom/react-week5-assignment-1/pull/98/commits/156e5fd96e816a378b18b8eab9994e9c8db9a594

jaeyooniverse committed:
Refactor #98 - actions.js

통일성을 위해 해당 파일에 존재하는 모든 함수를 화살표 함수로 리팩토링

해결: close #98

jaeyooniverse committed:
Refactor #98 - actions.test.js

불필요한 테스트 문구 제거

해결: close #98

jaeyooniverse committed:
Refactor #98 - reducer.js

코드를 간결하게 만들기 위해 actionTypes의 메소드 축약 표현으로 리팩토링

해결: close #98

jaeyooniverse committed:
Refactor #98 - reducer.test.js

BDD에서 Describe는 테스트 대상이기 때문에 reducer를 테스트 대상으로 정하면
it은 테스트의 행동이다. 그러므로 기존의 코드에서 액션 타입을 describe로
정했던 것을 제거하고, reducer에서 존재하지 않는 액션 타입은 특별한 예외,
심지어 많이 쓰이지도 않기 때문에 indent도 줄이고자 다른 테스트 대상으로 규정

해결: close #98

jaeyooniverse committed:
TEST: App.test.jsx

App 테스트 코드 추가

완료

jaeyooniverse committed:
FEAT - App 기능 추가

App 기능을 추가하고, import 에러 해결하기 위해 CategoriesContainer,
RegionsContainer, RestaurantsContainer export 설정

완료

jaeyooniverse committed:
TEST: RegionsContainer.test.jsx

RegionsContainer 테스트 코드 추가

완료

jaeyooniverse committed:
FEAT - RegionsContainer 기능 추가

RegionsContainer는 Regions의 상태를 관리하는 Container이다.

완료

jaeyooniverse committed:
TEST: Regions.test.jsx

Regions 테스트 코드 추

완료

hannut91 commented:
더 자세히 쓰려고 노력해 주셨군요 ㅎㅎ 내용도 작성해 주셨네요. 더 나아가서 진짜 의도를 드러내 보세요. 지금은 표면적인 이유만 작성된 것 같아요. 

> TEST: Regions.test.jsx
>
> Regions 테스트 코드 추

위의 커밋 메시지를 보면 제목만 봐서는 무엇을 하는건지 알기 어려워요. 커밋 작업 내용을 보니 저라면 이렇게 작성했을 것 같아요.

```
지역 선택시 선택됨이 표시되는지 테스트하라

지역 선택시 선택된 지역명에 (V)가 보여야 합니다. 
이를 확인할 수 있는 테스트를 추가했습니다.
```

hannut91 commented on src/App.test.jsx:
```diff
@@ -0,0 +1,47 @@
+
+  const renderApp = () => render(<App />);
+
+  it('"loadRegions"와 "loadCategories" 액션을 dispatch한다', () => {
+    renderApp();
+
+    expect(dispatch).toBeCalledTimes(2);
+  });
```
테스트에서는 loadRegions와 loadCategories를 테스트하는 것을 알 수 없어서 고민이 되네요. `redux-mock-store`를 사용하면 좀 더 자세히 테스트할 수 있는데 한 번 알아보세요

See also
  - https://github.com/reduxjs/redux-mock-store

hannut91 commented on src/containers/RegionsContainer.test.jsx:
```diff
@@ -0,0 +1,30 @@
+    regions,
+  }));
+
+  it('클릭한 버튼과 함께 "changeRegionName"을 dispatch한다', () => {
+    const { getByRole } = render(<RegionsContainer />);
+
+    const clickedButton = regions[0];
+
+    fireEvent.click(getByRole('button', { name: clickedButton.name }));
+
+    expect(dispatch).toBeCalledWith(changeRegionName(clickedButton));
+  });
```
지역을 클릭하면  "changeRegionName"을 dispatch한다 라고 작성하는 것은 어떨가요

jaeyooniverse committed:
FEAT - Regions 기능 추가

지역 버튼을 출력하는 기능인 Regions 기능 추가
Regions 하위에 Item 컴포넌트를 만들어 Categories에서도 재사용할 예정

완료

jaeyooniverse committed:
FEAT - Item 컴포넌트 추가

Regions에 해당하는 버튼과 Categories에 해당하는 버튼, 두 종류의 버튼이 있지만
UI 요소는 크게 다른 점이 없어서 Item 컴포넌트 하나를 만들어 Regions 버튼과
Categories 버튼을 만들 때 재활용한다.

완료

jaeyooniverse committed:
Refactor #98 - RegionsContainer.jsx

RegionsContainer 테스트가 게속 fail나서 RegionsContainer를 확인하던 중
changeRegionName action을 추가하지 않은 것을 확인함. 따라서,
action에 changeRegionName을 추가하고, changeRegionName prop에
변경할 regionName이 전달되어야 하므로 changeRegionName을
regionName으로 수정

완료

jaeyooniverse committed:
Refactor #98 - Regions.test.jsx

'region'버튼을 클릭하면 '(V)' 표시가 추가된다고 했는데 expect에 '(V)'를 넣지 않아서 수정함

완료

jaeyooniverse commented:
Commit 메시지를 작성하다가 에러 해결은 Refactor가 아닌 Fix라는 제목으로 작성했어야 했는데,
https://github.com/CodeSoom/react-week5-assignment-1/pull/98/commits/d8121484a1ee8151f17f2802ed418215b243a2c1 커밋 메시지를 작성하면서 아 맞다 이건 Refactor가 아닌거 같은데? 라고 인지했습니다.
다음 커밋 메시지 부터는 Fix라고 짓겠습니다!

jaeyooniverse committed:
Fix  #98-RegionsContainer.jsx

regionName을 인자로 받을 때, 9번째 줄 regionName이 이미 선언되어 있어서 중복된다는 eslint 에러 발생
따라서, regionName 인자를 regionNameParams라는 이름으로 수정함

완료

jaeyooniverse committed:
TEST: CategoriesContainer.test.jsx

Categories 컴포넌트의 상태를 관리하는 CategoriesContainer 컴포넌트의 테스트 코드
상태 관리 컴포넌트이므로 분류를 클릭하면 chagneCategoryId를 dispatch하는지 테스트한다

완료

jaeyooniverse committed:
FEAT - CategoriesContainer 기능 추가

Categories 컴포넌트의 상태를 관리하는 기능을 가진 CategoriesContainer 컴포넌트 추가

완료

jaeyooniverse committed:
TEST: Categories.test.jsx

Categories의 UI를 테스트하는 Categories.test.jsx 추가
category 버튼 렌더를 원활하게 하는지, 클릭했을 때 (V) 표시를 하는지 검사하는 테스트를 진행한다

완료

jaeyooniverse committed:
FEAT - Categories 기능 추가

Category 버튼의 UI를 담당하는 Category 컴포넌트 추가
Regions 컴포넌트에서 사용했던 Item을 재사용한다.

완료

jaeyooniverse committed:
Fix #98-CategoriesCotainer.test.jsx

CategoriesContainer 컴포넌트의의 import 경로를 RegionsContainer로 되어 있어서 CategoriesContaier로 변경

해결: #98

jaeyooniverse committed:
TEST: RestaurantsContainer.test.jsx

RestaurantsContainer 컴포넌트 테스트 코드
지역과 분류를 클릭했을 때 dispatch가 동작되는지 테스트한다.

완료

jaeyooniverse committed:
FEAT - RestaurantsContainer 기능 추가

지역과 분류 버튼을 클릭했을 때, 클릭한 지역 버튼의 name을 selectRegionName으로 받고
클릭한 분류 버튼의 id를 selectCategoryId로 받아, dispatch를 useEffect를 통해 진행한다.
조건에 해당하는 식당을 Restaurants 컴포넌트에서 출력한다.

완료

jaeyooniverse committed:
TEST: Restaurants.test.jsx

식당 이름을 제대로 출력하는지 테스트하는 코드를 가진 Restaurants.test.jsx를 추가한다.

완료

jaeyooniverse committed:
FEAT - Restaurants 기능 추가

RestaurantsContainer에서 받은 조건에 해당하는 식당들을 ul 태그 안 li로 출력한다.

완료

jaeyooniverse committed:
Fix #98- fetchRestaurants()

regionName과 categoryId를 initialState를 제외하고 감싸주는 객체가 또 존재하는 것이 아니므로
구조 분해 할당을 제거하여 api를 요청할 때 regionName과 categoryId가 undefined되는 오류를 해결한다.

완료#98

jaeyooniverse committed:
Refactor #98 - categoryId int to string

api를 요청할 때, 문자열로 보낼 때와 숫자를 보낼 때 차이가 없기 때문에 통일성을 위해
categoryId를 string으로 변경

완료

hannut91 commented:
음... 앞에 Fix, refactor, test를 붙이는 것 보다 더 중요한 것은 의도가 명확하게 드러나야 한다는 것입니다.

> FEAT - Restaurants 기능 추가
RestaurantsContainer에서 받은 조건에 해당하는 식당들을 ul 태그 안 li로 출력한다.

위의 커밋을 이렇게 아래와 같이 작성해야 합니다.

```
레스토랑 목록을 보여주어라

레스토랑 목록을 보여줍니다. 레스토랑에는 이름이 있는데, 이름을 보여주도록 했습니다.
레스토랑 목록을 여러개지만 순서가 없으니 <ul />태그와 <li/>태그를 이용해서 출력했습니다.
```

hannut91 commented on src/components/Categories.test.jsx:
```diff
@@ -0,0 +1,26 @@
+    });
+  });
+
+  it('"category"버튼을 클릭하면 "(V)"표시가 추가된다', () => {
+    const { getByText } = renderCategories(categories[0]);
+
+    expect(getByText(`${categories[0].name}(V)`)).toBeInTheDocument();
+  });
```
`클릭하면`이 테스트에서 찾아볼 수 없는 것 같아요. 테스트 대상을 더 명확히 작성해 보세요

jaeyooniverse replied:
category 버튼을 클릭하면 changeCategoryId action을 호출한 다음 (V) 표시를 추가하기 때문에 "category 버튼을 클릭하면" 과 같은 텍스트를 추가했는데 말씀하신 대로 fireEvent.Click과 같은 코드가 없으니 '"changeCategoryId를 호출하면 호출한 category 버튼에 "(V)"표시를 렌더한다'로 정정하도록 하겠습니다.

hannut91 commented on src/components/Item.jsx:
```diff
@@ -0,0 +1,13 @@
+        {name}
+        {selected && '(V)'}
+      </button>
+      <br />
```
이 안에 `<br />`이 있다는 것을 예상하기 어려운 것 같아요.

jaeyooniverse replied:
처음 리뷰를 봤을 때 br 태그가 있다는 것을 예상하기 어렵다는 말씀이 잘 와닿지가 않았는데, 과제의 예시처럼 버튼에 개행을 주기 위해 줬던 br태그를 지우고 해당 컴포넌트에 <></> 태그를 div로 바꿔보면서 아 나중에 기능을 추가하거나 보수할 일이 있을 때 br 태그를 쓰면 버튼에 왜 개행이 들어가있는지 예상하기 어려울 수 있겠구나라고 생각했습니다. 제 생각이 맞을까요?

hannut91 replied:
음... 이 컴포넌트를 쓰는 입장에서, button바로 아래에 개행이 있다는 것을 예측하기 어려울 것 같아요. 개행은 다른 곳에서 하는게 좋아보입니다

hannut91 commented on src/components/Item.jsx:
```diff
@@ -0,0 +1,13 @@
+export default function Item({ value, selected, onClick }) {
+  const { name } = value;
+
+  return (
+    <>
+      <button type="button" onClick={() => onClick(value)}>
+        {name}
+        {selected && '(V)'}
+      </button>
+      <br />
+    </>
+  );
```
value를 받아서 이름만 사용하고, 클릭했을 때 value를 그대로 돌려주고 있네요. 여기서 이름만 받도록하고, 클릭했을 때도 단순히 핸들러 함수를 실행만 하면 되도록 하는 것은 어떨까요? 그러면 버튼이 value자체에 대해서 몰라도 되서 더 좋을 것 같아요

hannut91 commented on src/components/Regions.test.jsx:
```diff
@@ -0,0 +1,23 @@
+    });
+  });
+
+  it('"region"버튼을 클릭하면 "(V)"표시가 추가된다', () => {
+    const { getByText } = renderRegions(regions[0]);
+
+    expect(getByText(`${regions[0].name}(V)`)).toBeInTheDocument();
+  });
```
여기도 테스트 이름을 다듬어야 할 것 같아요

hannut91 commented on src/components/Restaurants.jsx:
```diff
@@ -0,0 +1,9 @@
+export default function Restaurants({ restaurants }) {
+  return (
+    <ul>
+      {restaurants.map((restaurant) => (
+        <li key={restaurant.id}>{restaurant.name}</li>
+      ))}
+    </ul>
```
구조 분해 할당으로 단순하게 표현해 보세요

jaeyooniverse replied:
수정했습니다
done: https://github.com/CodeSoom/react-week5-assignment-1/pull/98/commits/c1526888d46058b20cc70c223423c540f0a50ee2

hannut91 commented on src/containers/CategoriesContainer.jsx:
```diff
@@ -0,0 +1,22 @@
+  const dispatch = useDispatch();
+  const { categories, categoryId } = useSelector((state) => state);
+
+  const handleChangeCategoryId = (categoryIdParams) => {
+    dispatch(changeCategoryId(categoryIdParams));
+  };
+
+  return (
+    <Categories
+      categories={categories}
+      changeCategoryId={categoryId}
+      onClick={handleChangeCategoryId}
+    />
```
핸들러 함수의 이름이 `handleChange`이군요. handleClick이 되어야 할 것 같아요

jaeyooniverse replied:
수정했습니다
done: https://github.com/CodeSoom/react-week5-assignment-1/pull/98/commits/a19ac4af6c4f09d08eb6cbbc2c635a3f8b5596ec

hannut91 commented on src/containers/CategoriesContainer.jsx:
```diff
@@ -0,0 +1,22 @@
+  const dispatch = useDispatch();
+  const { categories, categoryId } = useSelector((state) => state);
+
+  const handleChangeCategoryId = (categoryIdParams) => {
+    dispatch(changeCategoryId(categoryIdParams));
+  };
```
변수 이름에 왜 `Params`가 들어갔나요?

hannut91 commented on src/containers/RegionsContainer.jsx:
```diff
@@ -0,0 +1,24 @@
+  const dispatch = useDispatch();
+  const { regions, regionName } = useSelector((state) => state);
+
+  const handleChangeRegionName = (regionNameParams) => {
+    dispatch(changeRegionName(regionNameParams));
+  };
+
+  return (
+    <>
+      <Regions
+        regions={regions}
+        changeRegionName={regionName}
+        onClick={handleChangeRegionName}
+      />
+    </>
+  );
```
여기도 마찬가지로 params라는 변수명이 들어갔고, 핸들러 함수의 이름이 `handleChange`이군요. 그리고 `<Regions />`에 `changeRegionName`이 있는데, 변수명이 함수 이름처럼 생겨서 헷갈리네요.

jaeyooniverse replied:
처음에 reducer를 만들면서 지었던 이름들로 진행하며 점차 중복되는 이름들이 많아져 뜻이 헷갈리게 되는 결과가 나오게 되었습니다. 내일 대대적으로 이름들좀 손 보고 커밋한 다음 푸쉬하겠습니다.

jaeyooniverse replied:
손봤습니다!
done: https://github.com/CodeSoom/react-week5-assignment-1/pull/98/commits/da9ac16bbd949e0dadd2f4d37407e049cdc8a713

jaeyooniverse committed:
Refactor #98 - Categories.jsx, Item.jsx, Regions.jsx

기존 버튼들에 개행을 주기 위해 <br /> 태그를 이용하여 개행을 주었으나, 이렇게 코드를 작성하면
<br />이 존재하고 있다는 것을 예상하기 어려우므로 Categories, Item, Regions에 존재하는  <></>
태그를 div로 변경함으로써 리팩토링 함

완료

jaeyooniverse committed:
TEST - Categories.test.jsx

category 버튼을 클릭하면 changeCategoryId action을 호출한 다음 (V) 표시를 추가하기 때문에
'category 버튼을 클릭하면'과 같은 텍스트를 추가했는데 말씀하신 대로 fireEvent.Click과 같은
코드가 없으니 'changeCategoryId'를 호출하면 호출한 category 버튼에 '(V)' 표시를 렌더한다로
변경

완료

jaeyooniverse committed:
TEST - Regions.test.jsx

region 버튼을 클릭하면 changeRegionName action을 호출한 다음 (V) 표시를 추가하기 때문에
'region 버튼을 클릭하면'과 같은 텍스트를 추가했는데 말씀하신 대로 fireEvent.Click과 같은
코드가 없으니 'changeRegion'를 호출하면 호출한 지역 버튼에 '(V)' 표시를 렌더한다로
변경

완료

jaeyooniverse committed:
Refactor #98 - Restaurants.jsx

구조 분해 할당을 이용하여 기존 restaurant 요소에 restaurant.id restaurant.name을 이용했던 것을
{ id, name }으로 받아 id와 name으로 간단하게 만듬으로써 중복을 제거하고 코드를 간결하게 만듬

완료

jaeyooniverse committed:
Fix #98 - CategoriesContainer.jsx

버튼의 Click을 다루는 함수이므로 handleClick이 붙어야 한다.
따라서, 함수를 handleClickChangeCategoryId로 변경한다.

완료

jaeyooniverse committed:
Fix #98 - RegionsContainer.jsx

버튼의 Click을 다루는 함수이므로 handleClick이 붙어야 한다.
따라서, 함수를 handleClickChangeRegionName로 변경한다.

완료

jaeyooniverse committed:
Refactor regionName to selectedRegionName, categoryId to selectedCategoryId

reducer의 테스트 코드를 작성을 처음으로 진행하며 현재 버튼이 눌러진 지역과 분류를 regionName, categoryId라는
변수명으로 initialState에 담았는데, 개발을 진행하면서 기능이 추가될수록 비슷한 이름의 변수들과 함수, 인자
때문에 코드를 이해하는데 어려움을 겪게 됨. 따라서, 기존 regionName과 categoryId를 selecteRegionName,
selectedCategoryId로 변경함으로써 선택된이라는 표현을 추가함

완료

hannut91 commented:
커밋 메시지에 의도가 드러나도록 연습해 보세요

hannut91 commented on src/reducer.test.js:
```diff
@@ -0,0 +1,89 @@
+} from './actions';
+
+describe('reducer', () => {
+  it('sets regions with new regions', () => {
```
테스트 대상이 무엇인지 `describe`로 명시해 봅시다. 지금은 테스트 결과만 있어요. 만약 테스트가 실패했다고 메시지가 뜰 때, 무엇이 실패했는지 모를 것 같아요

jaeyooniverse committed:
Refactor #98 - reducer.test.js

reducer라는 파일에서 다루는 액션들을 테스트 대상으로 규정하여 기존에 테스트 대상을 추가하지 않은 채로
넣어놨던 it 코드들을 테스트 대상을 추가한 후 그 안에 넣어줬다.

완료

jaeyooniverse committed:
TEST: api.test.js

API 호출을 제대로 하는 지 테스트하라.

API를 호출하면 호출한 데이터가 미리 저장해 둔 fixtures의 데이터와 일치해야 합니다.
이를 확인할 수 있는 테스트를 추가했습니다.

완료

jaeyooniverse committed:
Refactor #98 - Item.jsx Button이 value값을 몰라도 되도록 변경

기존 코드는 value를 받아서 name만 사용하고 클릭했을 때 value를 그대로 또 되돌려주고 있기 때문에
이름만 받고 버튼이 value 자체에 대해서 몰라도 괜찮도록 handleClick이라는 단순 핸들러 함수를
만들어서 관심사를 분리한다.

완료

jaeyooniverse committed:
Chore: Github CI 통과를 위한 eslint 수정

CatgoriesContainer.test.jsx, index.jsx 파일에 존재하는 eslint 에러를 수정하여 Github CI를 통과

완료

hannut91 commented:
```
Chore: Github CI 통과를 위한 eslint 수정

CatgoriesContainer.test.jsx, index.jsx 파일에 존재하는 eslint 에러를 수정하여 Github CI를 통과
```

많이 좋아졌습니다. 하지만 아직입니다!! CI통과를 위해서가 아니라, 어떤 룰을 위해서 작업을 했는지 작성해 주셔야 합니다.

```
마지막 인자에 항상 콤마를 붙여라

마지막 인자에 콤마가 없으면, 새로운 라인을 추가하면 변경사항에 대해서 출력되고,
새로운 인자를 추가할 때 깜박하고 콤마를 추가안해서 에러가 나는 것을 방지할 수 있습니다.
React DOM에 렌더할 때 빠트린 콤마를 추가했습니다.

See also
  - https://eslint.org/docs/rules/comma-dangle
```

`진짜`의도에 대해서 더 더 더 생각해 보세요

hannut91 commented:
과제 풀이 영상을 참고해서 복습을 진행해 주세요. 특히 풀이 과정과 어떤 결정들을 해 나가는지 자세히 보시면 좋을 것 같아요
